kolibri/core/content/utils/search.py

```diff
@@ -89,27 +97,25 @@ def get_available_metadata_labels(base_queryset, limit_to_known_fields=True):
         hashlib.md5(str(base_queryset.query).encode("utf8")).hexdigest(),
     )
     if cache_key not in cache:
-        base_queryset = base_queryset.values("id").order_by()
-        queryset = base_queryset
-        all_values = []
-        if limit_to_known_fields:
-            lookup = get_all_contentnode_label_metadata()
-            lookup.pop("channels")
-            lookup.pop("languages")
-        else:
-            lookup = metadata_lookup
-        for field, values in lookup.items():
-            queryset = queryset.annotate(
-                **{
-                    value: Exists(base_queryset.has_all_labels(field, [value]))
-                    for value in values
-                }
-            )
-            all_values += values
-        result = queryset.values(*all_values).first()
+        base_queryset = base_queryset.order_by()
+        aggregates = {}
+        for field in bitmask_fieldnames:
+            field_agg = field + "_agg"
+            if connections[base_queryset.db].vendor == "sqlite":
```

Comment:
With the above `BitOr` potentially being `None`, perhaps this should also include `or BitOr is None`?